subprojects/core-api/src/main/java/org/gradle/api/initialization/dsl/VersionCatalogBuilder.java

```diff
@@ -150,4 +159,30 @@
         void withoutVersion();
     }
 
+    /**
+     * Allows configuring the version of a plugin
+     *
+     * @since 7.2
+     */
+    @Incubating
+    interface PluginAliasBuilder {
```

Comment:
I wonder if we should support a `withoutVersion` here which would allow to use 
```
plugins {
    alias(libs.myPlugin)
}
```
coupled with a version declaration in `settings.pluginManagement`

I understand it is more a corner case, but it might be good for being symmetric and complete.

Reply:
Closing as we agreed on not doing this now.

Reply:
After discussion, let's leave it out for now

Reply:
If I don't set the version in the TOML file, will it work or there is a check which blocks a plugin has no version in the TOML file?

I am adding the plugins to the implementation block inside buildSrc so I don't need to set a version, so I only need that the alias has no version, this is something I can do with "normal" dependencies (like kotlin test dependencies)

Reply:
Plugins declared in `implementation(...)` are regular dependencies with GAV coordinates, so this doesn't apply to them.

Reply:
Actually, the comment makes sense and is an aspect I had not considered: you will need the GAV to declare the plugin as an implementation in `buildSrc` but then you only need to plugin ID in the `plugins` block.
So maybe that support should be indeed added.
This is a stronger use case than the initial combination with `settings.pluginManagement` that I had in mind.

Reply:
I wonder if this wouldn't be more suited for the improvement to add:

```
implementation(pluginId("some.plugin"))
```

So that in the end, you can have both the coordinates _and_ the plugin id in the catalog?

Reply:
Sorry I was on mobile, I mean this use case:

```toml
# gradle/libs.versions.toml

[versions]
kotlin = "1.50.10"

kotlinGradlePlugin = { module = "org.jetbrains.kotlin:kotlin-gradle-plugin", version.ref = "kotlin" }

[plugins]
kotlinMultiplatform = {id = "org.jetbrains.kotlin.multiplatform" } # version is not indicated
```

```kotlin
// buildSrc/build.gradle.kts

dependencies {
    implementation(libs.kotlinGradlePlugin)
}
```

```kotlin
// someModule/build.gradle.kts

plugins {
     alias(libs.plugins.kotlinMultiplatform)
}
```

Reply:
If `implementation(pluginId("some.plugin"))` avoid having two declarations in the TOML file as I have used (one for the libraries and one for the plugins), it should be really great.

Reply:
@Vampire, your workaround to get catalogs working in precompiled plugins should work here too?

Reply:
You mean for doing `plugins { libs.plugins.foo }` in a precompiled script plugin?
I don't think so. The `plugins` block is extracted and applied separatly on a dummy project and we cannot influence that dummy project, so we cannot inject the hack-around.

Reply:
I mean this PR approach combined with your [workaround](https://github.com/gradle/gradle/issues/15383#issuecomment-779893192). I haven't tried if the accessors generated are accessible inside precompiled plugins with your workaround, so in `buildSrc` we can do:

```
// TOML file
[plugins]
kotlin="org.jetbrains.kotlin.jvm" // no version indicated

// buildSrc/src/main/kotlin/my-precompiled.gradle.kts
plugins {
    alias(libs.plugins.kotlin)
}
```

Reply:
I know what you meant and my answer applies.
`plugins` blocks of precompiled script plugins are extracted and applied separately on a dummy project and we cannot influence that dummy project, so we cannot inject the hack-around.
And as we cannot inject the hack-around, we cannot access `libs`.
And even if it were possible, `libs` is only available at runtime when the catalog is applied to the main project build, so additionally to the hack-around you would also need to declare the catalog on that dummy project, so that the `plugins` block can be evaluated for accessors generation.

Reply:
I tried to just create an alias without a version and it is not allowed, this is a limitation that avoids the usage of this feature because (overall in a multiprojects project) there is no need to set the version in all projects if you set it in the buildscript classpath in the root `build.gradle.kts` or in `buildSrc`.

```
  - Problem: In version catalog pluginLibs, on alias 'kotlinJvm' notation 'org.jetbrains.kotlin.jvm' is not a valid plugin notation.
    
    Reason: When using a string to declare plugin coordinates, you must use a valid plugin notation.
    
    Possible solution: Make sure that the coordinates consist of 2 parts separated by colons, eg: my.plugin.id:1.2.
```

Reply:
It forces to use a version, so there is no way to add a plugin alias without the version and I don't know how this feature can be used in a multi-projects project where you don't define a version in all projects, only in the root `build.gradle.kts` or `buildSrc/build.gradle.kts`